File: Python/6py 1/backwards_prime.py
"""
|--------------------------------------------------------------------------
| Backwards Read Primes
|--------------------------------------------------------------------------
|
"""

def is_prime(num):
   if num < 2: return False
   for i in range(2, int(num ** 0.5) + 1):
      if num % i == 0: return False
   return True
def reverse_number(num):
   return int(str(num)[::-1])
def backwards_prime(start, stop):
   return [num for num in range(start, stop + 1) if is_prime(num) and is_prime(reverse_number(num)) and num != reverse_number(num)]


def backwards_prime(start, stop):
   is_prime = lambda n : all(n % a for a in range(3, 1 + int(n ** 0.5), 2)) if n > 2 and n % 2 else n == 2
   return [n for n in range(start + ~start % 2, stop + 1, 2) if is_prime(n) and n != int(str(n)[::-1]) and is_prime(int(str(n)[::-1]))]


from gmpy2 import next_prime, is_prime
def backwards_prime(start, stop):
   res = []
   i = start
   while i<= stop :
      i = next_prime(i)
      if is_prime(int(str(i)[::-1])) and i<=stop and int(str(i)[::-1])!=i: res.append(int(i))
   return res


print(backwards_prime(9900, 10000), [9923, 9931, 9941, 9967])

File: PHP/6php 1/backwardsPrime.php
<?php
/*
|--------------------------------------------------------------------------
| Backwards Read Primes
|--------------------------------------------------------------------------
|
*/

function isPrime($num) {
   if ($num < 2) return false;
   for ($i = 2, $sqrt = sqrt($num); $i <= $sqrt; $i++) {
      if ($num % $i === 0) return false;
   }
   return true;
}
function reverseNumber($num) {
   return intval(strrev(strval($num)));
}
function backwardsPrime($start, $stop) {
   $result = array();
   for ($num = $start; $num <= $stop; $num++) {
      if (isPrime($num) && isPrime(reverseNumber($num)) && $num !== reverseNumber($num)) $result[] = $num;
   }
   return $result;
}


function backwardsPrime($start, $stop) {
   $isPrime = function ($n) {
      if ($n <= 2) return $n === 2;
      if (!($n % 2)) return false;
      foreach (range(3, sqrt($n), 2) as $i)
         if (!($n % $i)) return false;
      return true;
   };
   return array_values(array_filter(range($start, $stop), function ($n) use ($isPrime) {
      return $isPrime($n) && $n !== ($m = intval(strrev(strval($n)))) && $isPrime($m);
   }));
}


function isPrime(int $n): bool {
   if($n < 2) return false;
   for ($i = 2; $i <= sqrt($n); $i++){
     if ($n % $i == 0) return false;
   }
   return true;
}
function backwardsPrime($start, $stop){
   $primes = [];
   for ($i = $start; $i <= $stop; $i++){
      if (isPrime($i) && isPrime(strrev($i)) && ($i != strrev($i))) $primes[] = $i;
   }
   return $primes;
}


print_r(backwardsPrime(9900, 10000));